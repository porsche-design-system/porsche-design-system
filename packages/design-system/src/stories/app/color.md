# Color

Lorem ipsum.
![Color](../assets/color.png)