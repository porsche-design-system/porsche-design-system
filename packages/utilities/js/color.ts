export const color = {
  /**
   * Theme light as default theme
   */

  // Porsche brand colors
  brand: '#d5001c',

  // Default colors
  default: '#000',

  // Surface colors
  surface: '#f2f2f2',

  // Background colors
  background: '#fff',

  // Neutral colors
  neutralContrast: {
    high: '#323639',
    medium: '#626669',
    low: '#c9cacb',
  },

// Notification colors
  notifications: {
    error: '#018a16',
    warning: '#ff9b00',
    success: '#e00000',
  },

  // State colors
  state: {
    hover: '#d5001c',
    active: '#d5001c',
    focus: '#00d5b9',
    disabled: '#96989a',
  },
  /**
   * Theme dark as optional
   */
  darkTheme: {
    // Porsche brand colors
    brand: '#d5001c',

    // Default colors
    default: '#fff',

    // Surface colors
    surface: '#1a2023',

    // Background colors
    background: '#0e1418',

    // Neutral colors
    neutralContrast: {
      high: '#e3e4e5',
      medium: '#b0b1b2',
      low: '#4a4e51',
    },

// Notification colors
    notifications: {
      error: '#01ba1d',
      warning: '#ff9b00',
      success: '#fc1717',
    },

    // State colors
    state: {
      hover: '#d5001c',
      active: '#d5001c',
      focus: '#00d5b9',
      disabled: '#7c7f81',
    },
  },

  external: {
    facebook: "#1877f2",
    google: "#4285f4",
    instagram: "#e1306c",
    linkedin: "#0077b5",
    pinterest: '#e60023',
    twitter: "#1da1f2",
    wechat: "#1aad19",
    whatsapp: '#25d366',
    xing: '#006567',
    youtube: '#ff0000'
  }
};
