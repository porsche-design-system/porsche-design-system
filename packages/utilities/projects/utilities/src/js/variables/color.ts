export type Theme = 'light' | 'dark';

export const color = {
  /**
   * Theme light as default theme
   */
  brand: '#d5001c',
  default: '#000',
  background: {
    default: '#fff',
    surface: '#f2f2f2',
    shading: 'rgba(14, 20, 24, 0.9)'
  },
  neutralContrast: {
    high: '#323639',
    medium: '#626669',
    low: '#c9cacb'
  },
  notification: {
    success: '#018a16',
    warning: '#ff9b00',
    error: '#e00000',
    neutral: '#0061BD'
  },
  state: {
    hover: '#d5001c',
    active: '#d5001c',
    focus: '#00d5b9',
    disabled: '#96989a'
  },
  /**
   * Theme dark as optional
   */
  darkTheme: {
    brand: '#d5001c',
    default: '#fff',
    background: {
      default: '#0e1418',
      surface: '#1a2023',
      shading: 'rgba(14, 20, 24, 0.9)'
    },
    neutralContrast: {
      high: '#e3e4e5',
      medium: '#b0b1b2',
      low: '#4a4e51'
    },
    notification: {
      success: '#01ba1d',
      warning: '#ff9b00',
      error: '#fc1717',
      neutral: '#2193FF'
    },
    state: {
      hover: '#d5001c',
      active: '#d5001c',
      focus: '#00d5b9',
      disabled: '#7c7f81'
    }
  },
  /**
   * External brand colors
   */
  external: {
    facebook: '#1877f2',
    google: '#4285f4',
    instagram: '#e1306c',
    linkedin: '#0077b5',
    pinterest: '#e60023',
    twitter: '#1da1f2',
    wechat: '#1aad19',
    whatsapp: '#25d366',
    xing: '#006567',
    youtube: '#ff0000'
  }
};
