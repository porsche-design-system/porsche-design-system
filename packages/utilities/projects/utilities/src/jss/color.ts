export type ThemeDefault = 'light' | 'dark';
export type ThemeElectric = 'light-electric' | 'dark-electric';
export type Theme = ThemeDefault | ThemeElectric;

type ColorTheme = {
  brand: string;
  base: string;
  background: {
    base: string;
    surface: string;
    shading: string;
  };
  contrast: {
    high: string;
    medium: string;
    low: string;
  };
  notification: {
    success: string;
    successSoft: string;
    warning: string;
    warningSoft: string;
    error: string;
    errorSoft: string;
    neutral: string;
    neutralSoft: string;
  };
  state: {
    hover: string;
    active: string;
    focus: string;
    disabled: string;
  };
};

type ColorExternalKey =
  | 'facebook'
  | 'google'
  | 'instagram'
  | 'kakaotalk'
  | 'linkedin'
  | 'naver'
  | 'pinterest'
  | 'reddit'
  | 'tiktok'
  | 'twitter'
  | 'wechat'
  | 'whatsapp'
  | 'xing'
  | 'youtube';
type ColorExternal = { [key in ColorExternalKey]: string };

export const themeLight: ColorTheme = {
  brand: '#d5001c',
  base: '#000',
  background: {
    base: '#fff',
    surface: '#f2f2f2',
    shading: 'rgba(14, 20, 24, 0.9)',
  },
  contrast: {
    high: '#323639',
    medium: '#626669',
    low: '#e3e4e5',
  },
  notification: {
    success: '#018a16',
    successSoft: '#e5f3e7',
    warning: '#ff9b00',
    warningSoft: '#fff5e5',
    error: '#e00000',
    errorSoft: '#fae6e6',
    neutral: '#0061bd',
    neutralSoft: '#e5eff8',
  },
  state: {
    hover: '#d5001c',
    active: '#d5001c',
    focus: 'currentColor',
    disabled: '#96989a',
  },
};

export const themeDark: ColorTheme = {
  brand: '#d5001c',
  base: '#fff',
  background: {
    base: '#0e1418',
    surface: '#262b2e',
    shading: 'rgba(14, 20, 24, 0.9)',
  },
  // TODO: remove neutral?
  contrast: {
    high: '#e3e4e5',
    medium: '#b0b1b2',
    low: '#4a4e51',
  },
  notification: {
    success: '#01ba1d',
    successSoft: '#bfeec6',
    warning: '#ff9b00',
    warningSoft: '#ffe6bf',
    error: '#fc1717',
    errorSoft: '#fec5c5',
    neutral: '#2193ff',
    neutralSoft: '#c7e4ff',
  },
  state: {
    hover: '#ff0223',
    active: '#ff0223',
    focus: 'currentColor',
    disabled: '#7c7f81',
  },
};

export const themeLightElectric: ColorTheme = {
  ...themeLight,
  brand: '#00b0f4',
  state: {
    ...themeLight.state,
    hover: '#00b0f4',
    active: '#00b0f4',
  },
};

export const themeDarkElectric: ColorTheme = {
  ...themeDark,
  brand: '#00b0f4',
  state: {
    ...themeDark.state,
    hover: '#00b0f4',
    active: '#00b0f4',
  },
};

export const colorExternal: ColorExternal = {
  facebook: '#1877f2',
  google: '#4285f4',
  instagram: '#e1306c',
  kakaotalk: '#fae300',
  linkedin: '#0077b5',
  naver: '#03cf5d',
  pinterest: '#e60023',
  reddit: '#ff4500',
  tiktok: '#fe2c55',
  twitter: '#1da1f2',
  wechat: '#1aad19',
  whatsapp: '#25d366',
  xing: '#006567',
  youtube: '#ff0000',
};

export const color = {
  light: themeLight,
  dark: themeDark,
  'light-electric': themeLightElectric,
  'dark-electric': themeDarkElectric,
  external: colorExternal,
};
