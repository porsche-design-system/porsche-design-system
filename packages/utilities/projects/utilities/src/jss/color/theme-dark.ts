import type { ColorTheme } from './color-shared';

export const themeDark: ColorTheme = {
  brand: '#d5001c',
  base: '#fff',
  background: {
    base: '#0e1418',
    surface: '#262b2e',
    shading: 'rgba(14, 20, 24, 0.9)',
  },
  // TODO: remove neutral?
  contrast: {
    high: '#e3e4e5',
    medium: '#b0b1b2',
    low: '#4a4e51',
  },
  notification: {
    success: '#01ba1d',
    successSoft: '#bfeec6',
    warning: '#ff9b00',
    warningSoft: '#ffe6bf',
    error: '#fc1717',
    errorSoft: '#fec5c5',
    neutral: '#2193ff',
    neutralSoft: '#c7e4ff',
  },
  state: {
    hover: '#ff0223',
    active: '#ff0223',
    focus: 'currentColor',
    disabled: '#7c7f81',
  },
};
