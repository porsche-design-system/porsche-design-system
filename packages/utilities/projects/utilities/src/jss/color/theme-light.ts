import type { ColorTheme } from './shared';

export const themeLight: ColorTheme = {
  brand: '#d5001c',
  base: '#000',
  background: {
    base: '#fff',
    surface: '#f2f2f2',
    shading: 'rgba(14, 20, 24, 0.9)',
  },
  contrast: {
    high: '#323639',
    medium: '#626669',
    low: '#e3e4e5',
  },
  notification: {
    success: '#018a16',
    successSoft: '#e5f3e7',
    warning: '#ff9b00',
    warningSoft: '#fff5e5',
    error: '#e00000',
    errorSoft: '#fae6e6',
    neutral: '#0061bd',
    neutralSoft: '#e5eff8',
  },
  state: {
    hover: '#d5001c',
    active: '#d5001c',
    focus: 'currentColor',
    disabled: '#96989a',
  },
};
