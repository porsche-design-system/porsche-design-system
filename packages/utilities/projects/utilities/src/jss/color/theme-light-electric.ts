import type { ColorTheme } from './color-shared';
import { themeLight } from './theme-light';

export const themeLightElectric: ColorTheme = {
  ...themeLight,
  brand: '#00b0f4',
  state: {
    ...themeLight.state,
    hover: '#00b0f4',
    active: '#00b0f4',
  },
};
