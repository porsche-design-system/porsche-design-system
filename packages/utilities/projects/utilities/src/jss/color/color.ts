import { themeLightElectric } from './theme-light-electric';
import { themeLight } from './theme-light';
import { themeDarkElectric } from './theme-dark-electric';
import { themeDark } from './theme-dark';

type ThemeDefault = 'light' | 'dark';
type ThemeElectric = 'light-electric' | 'dark-electric';
type Theme = ThemeDefault | ThemeElectric;

type ColorExternalKey =
  | 'facebook'
  | 'google'
  | 'instagram'
  | 'kakaotalk'
  | 'linkedin'
  | 'naver'
  | 'pinterest'
  | 'reddit'
  | 'tiktok'
  | 'twitter'
  | 'wechat'
  | 'whatsapp'
  | 'xing'
  | 'youtube';

type ColorExternal = { [key in ColorExternalKey]: string };

const colorExternal: ColorExternal = {
  facebook: '#1877f2',
  google: '#4285f4',
  instagram: '#e1306c',
  kakaotalk: '#fae300',
  linkedin: '#0077b5',
  naver: '#03cf5d',
  pinterest: '#e60023',
  reddit: '#ff4500',
  tiktok: '#fe2c55',
  twitter: '#1da1f2',
  wechat: '#1aad19',
  whatsapp: '#25d366',
  xing: '#006567',
  youtube: '#ff0000',
};

const color = {
  light: themeLight,
  dark: themeDark,
  'light-electric': themeLightElectric,
  'dark-electric': themeDarkElectric,
  external: colorExternal,
};

export {
  color,
  colorExternal,
  ThemeDefault,
  ThemeElectric,
  Theme,
  themeLightElectric,
  themeLight,
  themeDarkElectric,
  themeDark,
};
