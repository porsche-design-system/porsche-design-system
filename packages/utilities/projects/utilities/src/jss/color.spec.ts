import { color, colorExternal, themeDark, themeDarkElectric, themeLight, themeLightElectric } from './color';

it('should contain correct values for themeLight', () => {
  expect(themeLight).toMatchSnapshot();
});

it('should contain correct values for themeDark', () => {
  expect(themeDark).toMatchSnapshot();
});

it('should contain correct values for themeLightElectric', () => {
  expect(themeLightElectric).toMatchSnapshot();
});

it('should contain correct values for themeDarkElectric', () => {
  expect(themeDarkElectric).toMatchSnapshot();
});

it('should contain correct values for colorExternal', () => {
  expect(colorExternal).toMatchSnapshot();
});

it('should contain correct values for color', () => {
  expect(color).toMatchSnapshot();
});
