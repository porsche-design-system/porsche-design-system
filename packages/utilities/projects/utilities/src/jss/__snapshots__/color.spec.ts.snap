// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should contain correct values for color 1`] = `
Object {
  "dark": Object {
    "background": Object {
      "base": "#0e1418",
      "shading": "rgba(14, 20, 24, 0.9)",
      "surface": "#262b2e",
    },
    "base": "#fff",
    "brand": "#d5001c",
    "neutralContrast": Object {
      "high": "#e3e4e5",
      "low": "#4a4e51",
      "medium": "#b0b1b2",
    },
    "notification": Object {
      "error": "#fc1717",
      "errorSoft": "#fec5c5",
      "neutral": "#2193ff",
      "neutralSoft": "#c7e4ff",
      "success": "#01ba1d",
      "successSoft": "#bfeec6",
      "warning": "#ff9b00",
      "warningSoft": "#ffe6bf",
    },
    "state": Object {
      "active": "#ff0223",
      "disabled": "#7c7f81",
      "focus": "currentColor",
      "hover": "#ff0223",
    },
  },
  "dark-electric": Object {
    "background": Object {
      "base": "#0e1418",
      "shading": "rgba(14, 20, 24, 0.9)",
      "surface": "#262b2e",
    },
    "base": "#fff",
    "brand": "#00b0f4",
    "neutralContrast": Object {
      "high": "#e3e4e5",
      "low": "#4a4e51",
      "medium": "#b0b1b2",
    },
    "notification": Object {
      "error": "#fc1717",
      "errorSoft": "#fec5c5",
      "neutral": "#2193ff",
      "neutralSoft": "#c7e4ff",
      "success": "#01ba1d",
      "successSoft": "#bfeec6",
      "warning": "#ff9b00",
      "warningSoft": "#ffe6bf",
    },
    "state": Object {
      "active": "#00b0f4",
      "disabled": "#7c7f81",
      "focus": "currentColor",
      "hover": "#00b0f4",
    },
  },
  "external": Object {
    "facebook": "#1877f2",
    "google": "#4285f4",
    "instagram": "#e1306c",
    "kakaotalk": "#fae300",
    "linkedin": "#0077b5",
    "naver": "#03cf5d",
    "pinterest": "#e60023",
    "reddit": "#ff4500",
    "tiktok": "#fe2c55",
    "twitter": "#1da1f2",
    "wechat": "#1aad19",
    "whatsapp": "#25d366",
    "xing": "#006567",
    "youtube": "#ff0000",
  },
  "light": Object {
    "background": Object {
      "base": "#fff",
      "shading": "rgba(14, 20, 24, 0.9)",
      "surface": "#f2f2f2",
    },
    "base": "#000",
    "brand": "#d5001c",
    "neutralContrast": Object {
      "high": "#323639",
      "low": "#e3e4e5",
      "medium": "#626669",
    },
    "notification": Object {
      "error": "#e00000",
      "errorSoft": "#fae6e6",
      "neutral": "#0061bd",
      "neutralSoft": "#e5eff8",
      "success": "#018a16",
      "successSoft": "#e5f3e7",
      "warning": "#ff9b00",
      "warningSoft": "#fff5e5",
    },
    "state": Object {
      "active": "#d5001c",
      "disabled": "#96989a",
      "focus": "currentColor",
      "hover": "#d5001c",
    },
  },
  "light-electric": Object {
    "background": Object {
      "base": "#fff",
      "shading": "rgba(14, 20, 24, 0.9)",
      "surface": "#f2f2f2",
    },
    "base": "#000",
    "brand": "#00b0f4",
    "neutralContrast": Object {
      "high": "#323639",
      "low": "#e3e4e5",
      "medium": "#626669",
    },
    "notification": Object {
      "error": "#e00000",
      "errorSoft": "#fae6e6",
      "neutral": "#0061bd",
      "neutralSoft": "#e5eff8",
      "success": "#018a16",
      "successSoft": "#e5f3e7",
      "warning": "#ff9b00",
      "warningSoft": "#fff5e5",
    },
    "state": Object {
      "active": "#00b0f4",
      "disabled": "#96989a",
      "focus": "currentColor",
      "hover": "#00b0f4",
    },
  },
}
`;

exports[`should contain correct values for colorExternal 1`] = `
Object {
  "facebook": "#1877f2",
  "google": "#4285f4",
  "instagram": "#e1306c",
  "kakaotalk": "#fae300",
  "linkedin": "#0077b5",
  "naver": "#03cf5d",
  "pinterest": "#e60023",
  "reddit": "#ff4500",
  "tiktok": "#fe2c55",
  "twitter": "#1da1f2",
  "wechat": "#1aad19",
  "whatsapp": "#25d366",
  "xing": "#006567",
  "youtube": "#ff0000",
}
`;

exports[`should contain correct values for themeDark 1`] = `
Object {
  "background": Object {
    "base": "#0e1418",
    "shading": "rgba(14, 20, 24, 0.9)",
    "surface": "#262b2e",
  },
  "base": "#fff",
  "brand": "#d5001c",
  "neutralContrast": Object {
    "high": "#e3e4e5",
    "low": "#4a4e51",
    "medium": "#b0b1b2",
  },
  "notification": Object {
    "error": "#fc1717",
    "errorSoft": "#fec5c5",
    "neutral": "#2193ff",
    "neutralSoft": "#c7e4ff",
    "success": "#01ba1d",
    "successSoft": "#bfeec6",
    "warning": "#ff9b00",
    "warningSoft": "#ffe6bf",
  },
  "state": Object {
    "active": "#ff0223",
    "disabled": "#7c7f81",
    "focus": "currentColor",
    "hover": "#ff0223",
  },
}
`;

exports[`should contain correct values for themeDarkElectric 1`] = `
Object {
  "background": Object {
    "base": "#0e1418",
    "shading": "rgba(14, 20, 24, 0.9)",
    "surface": "#262b2e",
  },
  "base": "#fff",
  "brand": "#00b0f4",
  "neutralContrast": Object {
    "high": "#e3e4e5",
    "low": "#4a4e51",
    "medium": "#b0b1b2",
  },
  "notification": Object {
    "error": "#fc1717",
    "errorSoft": "#fec5c5",
    "neutral": "#2193ff",
    "neutralSoft": "#c7e4ff",
    "success": "#01ba1d",
    "successSoft": "#bfeec6",
    "warning": "#ff9b00",
    "warningSoft": "#ffe6bf",
  },
  "state": Object {
    "active": "#00b0f4",
    "disabled": "#7c7f81",
    "focus": "currentColor",
    "hover": "#00b0f4",
  },
}
`;

exports[`should contain correct values for themeLight 1`] = `
Object {
  "background": Object {
    "base": "#fff",
    "shading": "rgba(14, 20, 24, 0.9)",
    "surface": "#f2f2f2",
  },
  "base": "#000",
  "brand": "#d5001c",
  "neutralContrast": Object {
    "high": "#323639",
    "low": "#e3e4e5",
    "medium": "#626669",
  },
  "notification": Object {
    "error": "#e00000",
    "errorSoft": "#fae6e6",
    "neutral": "#0061bd",
    "neutralSoft": "#e5eff8",
    "success": "#018a16",
    "successSoft": "#e5f3e7",
    "warning": "#ff9b00",
    "warningSoft": "#fff5e5",
  },
  "state": Object {
    "active": "#d5001c",
    "disabled": "#96989a",
    "focus": "currentColor",
    "hover": "#d5001c",
  },
}
`;

exports[`should contain correct values for themeLightElectric 1`] = `
Object {
  "background": Object {
    "base": "#fff",
    "shading": "rgba(14, 20, 24, 0.9)",
    "surface": "#f2f2f2",
  },
  "base": "#000",
  "brand": "#00b0f4",
  "neutralContrast": Object {
    "high": "#323639",
    "low": "#e3e4e5",
    "medium": "#626669",
  },
  "notification": Object {
    "error": "#e00000",
    "errorSoft": "#fae6e6",
    "neutral": "#0061bd",
    "neutralSoft": "#e5eff8",
    "success": "#018a16",
    "successSoft": "#e5f3e7",
    "warning": "#ff9b00",
    "warningSoft": "#fff5e5",
  },
  "state": Object {
    "active": "#00b0f4",
    "disabled": "#96989a",
    "focus": "currentColor",
    "hover": "#00b0f4",
  },
}
`;
